vignettes/deployment.Rmd

```diff
@@ -0,0 +1,188 @@
+
+    By default this stage looks like
+
+    ```{r eval = FALSE}
+    if (Sys.getenv("BUILD_PKGDOWN") != "") {
+      get_stage("before_deploy") %>%
+        add_step(step_setup_ssh())
+    }
+    ```
+
+    This block calls the function `step_setup_ ssh()` if the environment variable `"BUILD_PKGDOWN"` is set in the Travis build.
+    That is just an example of an option that can be set to run certain commands (or stages) conditionally (see [here](advanced.html#running-stages-conditionally) for more examples).
+    After the prerequisite of setting up an ssh key for the upcoming deployment has been finished, 
+
+2. Travis is told to use a "script" (provider: script) for the deployment (which holds further instructions).
+
+3. This "script" is calling `tic::deploy()`.
+
+4. All this is happening on every branch. If you want to restrict it to certain branches only, you need to set a condition in `tic.R`.
+
+    What happens now is that `tic::deploy()` again searches in `tic.R` for the "deploy" stage and then runs whatever is specified there.
+
+    With the default template of `tic.R`, the following will be executed:
+
+    ```{r eval = FALSE}
+    get_stage("deploy") %>%
+      add_step(step_build_pkgdown()) %>% 
+      add_step(step_push_deploy())
+    ```
+
+    First, `step_build_pkgdown()` will build your _pkgdown_ site and afterwards (note the `pipe` operator chaining the commands), `step_push_deploy()` takes care pushing the results to the repo.
+    By default this will be the `docs/` directory of the `master` branch.
+    
+### Deploying to docs/ (master) or gh-pages branch
+
+#### master branch
+
+By default the deployment is done to the `docs/` directory of the `master` branch.
+This option has the following advantages:
+
+- You can have per-branch versions of your _pkgdown_ site
+- Per-branch versions enable the possibility to have preview for pull requests via https://www.netlify.com/
+
+A disadvantage is that the `master` branch will be cluttered by automatic commits triggered by Travis that push the changes of the _pkgdown_ site to the `master` branch.
+
+#### gh-pages branch
+
+You can optionally set the deployment branch to `gh-pages` and then change the branch serving your site in the GitHub repo settings (change it to `gh-pages`).
+This option has the advantage 
+
+- to not have all the automatic commits from Travis in the `master` branch -> the _pkgdown_ site will be "silently" updated in the background in the `gh-pages` branch.
+
+A disadvantage is that you cannot have per-branch versions of your site and hence no pull request previews of the site.
+
+To use this option with *tic*, specify `branch = "gh-pages", path = "docs"` in `step_push_deploy()`.
+As the site serving in this setting will use the root of the branch, the `gh-pages` branch will only consist of the public _pkgdown_ files used to serve the site. 
+
+### Conditional deployment
```

Comment:
I still don't understand this section. I think we can point to Travis CI documentation for restricting deployment and for multi-stage build, and describe only "our" way here.

Reply:
Rephrased it and focused more on the _tic_ way: https://github.com/ropenscilabs/tic/pull/80/commits/2a72bb9ffc51170db9b79de8b6595dacf3a766ec.